drivers/azure/azureutil/azureutil.go

```diff
@@ -520,6 +532,33 @@ func (a AzureClient) CreateVirtualMachine(resourceGroup, name, location, size, a
 	return err
 }
 
+func (a AzureClient) CreateVirtualMachineExtension(os, resourceGroup, name, location string) error {
+	vmExtension := compute.VirtualMachineExtension{
+		Type:     to.StringPtr("Microsoft.Compute/virtualMachines/extensions"),
+		Name:     to.StringPtr("winrm"),
+		Location: to.StringPtr(location),
+		Properties: &compute.VirtualMachineExtensionProperties{
+			Publisher:               to.StringPtr("Microsoft.Compute"),
+			Type:                    to.StringPtr("CustomScriptExtension"),
+			TypeHandlerVersion:      to.StringPtr("1.0"),
+			AutoUpgradeMinorVersion: to.BoolPtr(true),
+			Settings: &map[string]interface{}{
+				"fileUris": []string{
+					"https://raw.githubusercontent.com/docker/machine/master/vendor/github.com/Azure/azure-quickstart-templates/201-vm-winrm-windows/ConfigureWinRM.ps1",
```

Comment:
I guess when you correctly run godep, these files will no longer be here. So we need to have a discussion where to check in these 3 files. I think a directory under `contrib` would be fine.

Also as we keep fetching these from github, users may run into throttling as github aggressively throttles raw file downloads. Perhaps if docker can offer a better location to store these, that would be the best solution.